ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -7061,7 +7061,9 @@ BoundRow * HqlCppTranslator::bindSelectorAsSelf(BuildCtx & ctx, IReferenceSelect
     }
 
     //Need to bind a delta address to a new variable.
-//  throwUnexpected();  // check this is actually called
+    if (!rootRow->queryBuilder())
```

Comment:
This is backup protection to throw an error instead of coring if there is another instance of the same issue.  (See line 7080 just below the end of the context extract)

     generateExprCpp(s, rootRow->queryBuilder()).append(",");

which was the place the code was actually crashing.